# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1452055071 -39600
#      Wed Jan 06 15:37:51 2016 +1100
# Node ID 10071454a4ba168319c8c15af884152d023922ca
# Parent  0967e239338608afa2679aafbab76c070eda6918
Allow assigment from "float complex matrix" to "complex matrix" (bug #46816)

* libinterp/operators/op-fcm-fcm.cc (install_fcm_fcm_ops) : Added  dbl_assign
  operator, and renamd existing dbl_assign to sgl_assign in keeping with the
  naming convention "dbl" is for a double destination, "sgl" for float dest.

diff --git a/libinterp/operators/op-fcm-fcm.cc b/libinterp/operators/op-fcm-fcm.cc
--- a/libinterp/operators/op-fcm-fcm.cc
+++ b/libinterp/operators/op-fcm-fcm.cc
@@ -209,20 +209,22 @@ DEFNDCATOP_FN (fcm_fcm, float_complex_ma
 DEFNDCATOP_FN (cm_fcm, complex_matrix, float_complex_matrix,
                float_complex_array, float_complex_array, concat)
 
 DEFNDCATOP_FN (fcm_cm, float_complex_matrix, complex_matrix,
                float_complex_array, float_complex_array, concat)
 
 DEFNDASSIGNOP_FN (assign, float_complex_matrix, float_complex_matrix,
                   float_complex_array, assign)
-DEFNDASSIGNOP_FN (dbl_clx_assign, float_complex_matrix, complex_matrix,
+DEFNDASSIGNOP_FN (sgl_clx_assign, float_complex_matrix, complex_matrix,
                   float_complex_array, assign)
-DEFNDASSIGNOP_FN (dbl_assign, float_complex_matrix, matrix,
+DEFNDASSIGNOP_FN (sgl_assign, float_complex_matrix, matrix,
                   float_complex_array, assign)
+DEFNDASSIGNOP_FN (dbl_assign, complex_matrix, float_complex_matrix,
+                  complex_array, assign)
 
 DEFNULLASSIGNOP_FN (null_assign, float_complex_matrix, delete_elements)
 
 DEFNDASSIGNOP_OP (assign_add, float_complex_matrix,
                   float_complex_matrix, float_complex_array, +=)
 DEFNDASSIGNOP_OP (assign_sub, float_complex_matrix,
                   float_complex_matrix, float_complex_array, -=)
 DEFNDASSIGNOP_FNOP (assign_el_mul, float_complex_matrix, float_complex_matrix,
@@ -306,19 +308,21 @@ install_fcm_fcm_ops (void)
   INSTALL_CATOP (octave_complex_matrix,
                  octave_float_complex_matrix, cm_fcm);
   INSTALL_CATOP (octave_float_complex_matrix,
                  octave_complex_matrix, fcm_cm);
 
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
                     octave_float_complex_matrix, assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
-                    octave_complex_matrix, dbl_clx_assign);
+                    octave_complex_matrix, sgl_clx_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
-                    octave_matrix, dbl_assign);
+                    octave_matrix, sgl_assign);
+  INSTALL_ASSIGNOP (op_asn_eq, octave_complex_matrix,
+                    octave_float_complex_matrix, dbl_assign);
 
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
                     octave_null_matrix, null_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
                     octave_null_str, null_assign);
   INSTALL_ASSIGNOP (op_asn_eq, octave_float_complex_matrix,
                     octave_null_sq_str, null_assign);
 
